src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,174 @@
+            }
+            size++;
+            return;
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> tempNode = findNodeByIndex(index);
+        Node<T> previous = tempNode.prev;
+        previous.next = node;
+        tempNode.prev = node;
+        node.prev = previous;
+        node.next = tempNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        Node<T> tempNode = findNodeByIndex(index);
+        return tempNode.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexCheck(index);
+        Node<T> node = new Node<>(value);
+        T returnedValue;
+        if (index == 0) {
+            node.prev = null;
+            node.next = first.next;
+            returnedValue = first.item;
+            first = node;
+            return returnedValue;
+        }
+        Node<T> tempNode = findNodeByIndex(index);
+        returnedValue = tempNode.item;
+        tempNode.item = node.item;
+        return returnedValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index);
+        T removedItem;
+        if (index == 0) {
+            if (first.next == null) {
+                removedItem = first.item;
+                first = null;
+                size = 0;
+                return removedItem;
+            }
+            first.next.prev = null;
+            removedItem = first.item;
+            first = first.next;
+            size--;
+            return removedItem;
+        }
+        if (index == size - 1) {
+            last.prev.next = null;
+            removedItem = last.item;
+            last = last.prev;
+            size--;
+            return removedItem;
+        }
+        Node<T> tempNode = findNodeByIndex(index);
+        removedItem = tempNode.item;
+        tempNode.prev.next = tempNode.next;
+        tempNode.next.prev = tempNode.prev;
+        size--;
+        return removedItem;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> tempNode = first;
+        for (int i = 0; i < size; i++) {
+            if (tempNode.item == object || (object != null && object.equals(tempNode.item))) {
+                remove(i);
+                return true;
+            }
+            tempNode = tempNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == ZERO;
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        Node<T> tempNode = first;
+        for (int i = 0; i < index; i++) {
+            tempNode = tempNode.next;
+        }
```

Comment:
if the index is in the first part of list, we can begin the movement from the head, otherwise from the tail